### PR TITLE
Feature drivetrain

### PR DESCRIPTION
**Pull request by rohitchawla28**
Subsystem is complete for all manual functionalities.

### CODE REVIEW

File: Competition/.vscode/settings.json [hunk 1]
```diff
@@ -14,5 +14,68 @@
     "**/.factorypath": true,
     "**/*~": true
   },
-  "C_Cpp.default.configurationProvider": "vscode-wpilib"
+  "C_Cpp.default.configurationProvider": "vscode-wpilib",
```
Comment (mray190) [inline Competition/.vscode/settings.json:17]: I don't know if we should check this in....
This seems user specific?

File: Competition/src/main/cpp/Drivetrain.cpp [hunk 2]
```diff
@@ -1,217 +1,229 @@
-/*----------------------------------------------------------------------------*/
```
Comment (mray190) [inline Competition/src/main/cpp/Drivetrain.cpp:1]: Keep copyright and license header

File: Competition/src/main/cpp/Drivetrain.cpp [hunk 3]
```diff
@@ -1,217 +1,229 @@
...
-/* Open Source Software - may be modified and shared by FRC teams. The code   */
-/* must be accompanied by the FIRST BSD license file in the root directory of */
-/* the project.                                                               */
-/*----------------------------------------------------------------------------*/
-
 #include "Drivetrain.h"
 
-Drivetrain::Drivetrain() : boostMultiplier{DriveConstants::kNoBoost},
-                           kDriveKinematics{DriveConstants::kTrackwidth},
+Drivetrain::Drivetrain() : kDriveKinematics{DriveConstants::kTrackwidth},
```
Comment (mray190) [inline Competition/src/main/cpp/Drivetrain.cpp]: Call ValorSubsystem constructor in initializer list to kick off init function

File: Competition/src/main/cpp/Drivetrain.cpp [hunk 4]
```diff
@@ -1,217 +1,229 @@
...
-  InitDrives(rev::CANSparkMax::IdleMode::kCoast);
+                           leftDriveLead{DriveConstants::CAN_ID_LEFT_LEAD, rev::CANSparkMax::MotorType::kBrushless},
+                           leftDriveFollowA{DriveConstants::CAN_ID_LEFT_FOLLOW_A, rev::CANSparkMax::MotorType::kBrushless},
+                           leftDriveFollowB{DriveConstants::CAN_ID_LEFT_FOLLOW_B, rev::CANSparkMax::MotorType::kBrushless},
+                           rightDriveLead{DriveConstants::CAN_ID_RIGHT_LEAD, rev::CANSparkMax::MotorType::kBrushless},
+                           rightDriveFollowA{DriveConstants::CAN_ID_RIGHT_FOLLOW_A, rev::CANSparkMax::MotorType::kBrushless},
+                           rightDriveFollowB{DriveConstants::CAN_ID_RIGHT_FOLLOW_B, rev::CANSparkMax::MotorType::kBrushless},
+                           odometry{frc::Rotation2d(units::degree_t(getHeading()))},
+                           driverController(NULL) {
+    kTrajectoryConfigF.SetKinematics(kDriveKinematics);
```
Comment (mray190) [inline Competition/src/main/cpp/Drivetrain.cpp]: Move all of these setup calls into the init function

File: Competition/src/main/include/Drivetrain.h [hunk 5]
```diff
@@ -1,124 +1,101 @@
-/*----------------------------------------------------------------------------*/
```
Comment (mray190) [inline Competition/src/main/include/Drivetrain.h:1]: Keep copyright and license header

**Review by mray190 — CHANGES_REQUESTED** [review]

File: Competition/.vscode/settings.json (hunk 1 above)
Comment (rohitchawla28) [reply Competition/.vscode/settings.json:17]: yeah...i have no clue what this is

Comment (rohitchawla28) [reply Competition/.vscode/settings.json:17]: should i just delete it or is there a way to not add the changes to this file

Comment (mray190) [reply Competition/.vscode/settings.json:17]: mehhh it should be fine. I dont really want to deal with it lol. we can leave it

**Review by mray190 — APPROVED** [review]

**Pull request merged**